python/ray/train/v2/_internal/state/schema.py

```diff
@@ -214,6 +214,47 @@ class DecoratedTrainRunAttempt(TrainRunAttempt):
     )
 
 
+@DeveloperAPI
+class DatasetsDetails(BaseModel):
+    """Datasets details for a Train run, including dataset names and data config."""
+
+    datasets: List[str] = Field(description="A list of dataset names for a Train run.")
+    data_config: Dict[str, Any] = Field(description="The data config for a Train run.")
+
+
+@DeveloperAPI
+class RuntimeConfiguration(BaseModel):
+    """Runtime configuration parameters for a Train run, encompassing failure,
+    runtime environment, checkpoint settings, and storage path."""
+
+    failure_config: Dict[str, Any] = Field(
+        description="The failure config for a Train run."
+    )
+    worker_runtime_env: Dict[str, Any] = Field(
+        description="The worker runtime env for a Train run."
+    )
+    checkpoint_config: Dict[str, Any] = Field(
+        description="The checkpoint config for a Train run."
+    )
+    storage_path: str = Field(description="The storage path for a Train run.")
```

Comment:
We should try to have typed schemas wherever possible so that there is a strict interface that the consumer can read from.